test(napi/oxlint): disable watch mode in `pnpm test`

overlookmotel opened the pull request:
We now run tests against the build in `dist`, so running tests in watch mode is misleading - the code being tested doesn't update until you run `pnpm run build-test`.

overlookmotel commented:
* **#12617** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/12617?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/12617?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=12617).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

camc314 commented on napi/oxlint2/package.json:
```diff
@@ -11,7 +11,7 @@
     "build-napi-release": "pnpm run build-napi --release",
     "build-js": "node scripts/build.js",
-    "test": "tsc && vitest"
+    "test": "tsc && vitest --dir ./test run"
```
```suggestion
    "test": "tsc && vitest --dir ./test run",
    "test:watch": "vitest"

overlookmotel replied:
I agree, watch mode is nice. But, unless I'm misunderstanding, when you update a file, tests will re-run, but `e2e` tests are still running against the *old* code because that's what `dist` still contains. So it's not really watching your changes at all.

camc314 replied:
oh you're right

let's leave for now

camc314 replied:
there probably is a vitest setting, but i think it's not a priority to make work rn

overlookmotel committed:
test(napi/oxlint): disable watch mode in `pnpm test` (#12617)

We now run tests against the build in `dist`, so running tests in watch mode is misleading - the code being tested doesn't update until you run `pnpm run build-test`.
